irods_browser_app/views.py

```diff
@@ -0,0 +1,137 @@
+import json
+import os
+from django.http import HttpResponse, HttpResponseRedirect
+from irods.session import iRODSSession
+from irods.exception import CollectionDoesNotExist
+from hs_core import hydroshare
+
+irods_sess = None
+
+def search_ds(coll):
+    store = {}
+    file = []
+    folder = []
+    if coll.data_objects:
+        for files in coll.data_objects:
+            file.append(files.name)
+    if coll.subcollections:
+        for folders in coll.subcollections:
+            folder.append(folders.name)
+
+    store['files'] = file
+    store['folder'] = folder
+    return store
+
+# Create your views here.
+def login(request):
+    if request.method == 'POST':
+        port = int(request.POST['port'])
+        user = str(request.POST['user'])
+        password = str(request.POST['password'])
+        zone = str(request.POST['zone'])
+        host = str(request.POST['host'])
+        datastore = "/%s/home/%s" % (zone, user)
+
+        response_data = {}
+
+        global irods_sess
+        irods_sess = iRODSSession(user=user, password=password, zone=zone, host=host, port=port)
```

Comment:
I was trying to avoid this global session object as well but could not find a way to make it work. Could you elaborate a bit on your last point on possible alternative approach? To send the credentials in each operation that uses them, I have to store the credentials in memory or sessions to have them being passed around since multiple operations involve multiple views, right?